render/src/html/render.rs

```diff
@@ -0,0 +1,202 @@
+        _context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        // TODO: improve handling of attributes
+        // let attributes = serde_json::from_str::<VerbatimAttributes>(
+        //     &verbatim.attributes.as_ref().cloned().unwrap_or_default(),
+        // )
+        // .ok();
+
+        // let language = match attributes.as_ref() {
+        //     Some(attrs) => attrs.language.clone().unwrap_or(PLAIN_SYNTAX.to_string()),
+        //     None => PLAIN_SYNTAX.to_string(),
+        // };
+        let language = "rust";
+
+        let inner = Html::with(
+            HtmlHead {
+                syntax_highlighting_used: true,
+                ..Default::default()
+            },
+            HtmlBody::from(HtmlElement {
+                name: "code".to_string(),
+                attributes: HtmlAttributes::default(),
+                content: Some(
+                    highlight::highlight_content(&verbatim.content, language)
+                        .unwrap_or(verbatim.content.clone()),
+                ),
+            }),
+        );
+
+        Ok(Html::nested("pre", HtmlAttributes::default(), inner))
+    }
+
+    fn render_bold(
+        &mut self,
+        bold: &unimarkup_inline::NestedContent,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_nested_inline(bold, context)?;
+
+        Ok(Html::nested("strong", HtmlAttributes::default(), inner))
+    }
+
+    fn render_italic(
+        &mut self,
+        italic: &unimarkup_inline::NestedContent,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_nested_inline(italic, context)?;
+
+        Ok(Html::nested("em", HtmlAttributes::default(), inner))
+    }
+
+    fn render_underline(
+        &mut self,
+        underline: &unimarkup_inline::NestedContent,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_nested_inline(underline, context)?;
+        let mut attributes = HtmlAttributes::default();
+        attributes.push(HtmlAttribute {
+            name: "style".to_string(),
+            value: Some("text-decoration: underline;".to_string()),
+        });
+
+        Ok(Html::nested("span", attributes, inner))
+    }
+
+    fn render_subscript(
+        &mut self,
+        subscript: &unimarkup_inline::NestedContent,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_nested_inline(subscript, context)?;
+
+        Ok(Html::nested("sub", HtmlAttributes::default(), inner))
+    }
+
+    fn render_superscript(
+        &mut self,
+        superscript: &unimarkup_inline::NestedContent,
+        context: &Context,
+    ) -> Result<Html, crate::log_id::RenderError> {
+        let inner = self.render_nested_inline(superscript, context)?;
+
+        Ok(Html::nested("sup", HtmlAttributes::default(), inner))
```

Comment:
Another improvement we could have is to use `enum HtmlTag` instead of strings for tags. I'll leave it up to you to decide whether we should do it in this PR or in another one (or at all).

Reply:
good idea. I will implement it in this PR